internal/resolver/mtaresolver.go

```diff
@@ -35,17 +33,13 @@ type ResolveResult struct {
 }
 
 // Resolve - resolve module's parameters
-func Resolve(workspaceDir, moduleName, path string, envFile string) (result ResolveResult, err error) {
+func Resolve(workspaceDir, moduleName, path string, extensions []string, envFile string) (result ResolveResult, messages []string, err error) {
 	if len(moduleName) == 0 {
-		return result, errors.New(emptyModuleNameMsg)
+		return result, nil, errors.New(emptyModuleNameMsg)
 	}
-	yamlData, err := ioutil.ReadFile(path)
+	mtaRaw, messages, err := mta.GetMtaFromFile(path, extensions)
 	if err != nil {
-		return result, errors.Wrapf(err, pathNotFoundMsg, path)
-	}
-	mtaRaw, err := mta.Unmarshal(yamlData)
-	if err != nil {
-		return result, errors.Wrapf(err, unmarshalFailsMsg, path)
+		return result, messages, errors.Wrapf(err, unmarshalFailsMsg, path)
```

Comment:
I think unmarshalFailsMsg message is not relevant now.

Reply:
The message is still relevant, I think - we want a good error message so the user will understand the issue. Without it, when unmarshal fails, the user will get messages like:
` yaml: unmarshal errors:\n  line 7: field type1 not found in type mta.Module`
I improved the message text and changed it message to be returned from `GetMtaFromFile` so we will always show it.